From a8506d949c5dd8b4e4d743d2ec79cd014e6b31ce Mon Sep 17 00:00:00 2001
From: QiuJi <qiuji@iscas.ac.cn>
Date: Sat, 28 Aug 2021 10:29:16 +0800
Subject: [PATCH] [riscv64][sparkplug] Clobber accumulator in StaGlobal

Port: 732f394c5d88862ccb20a79eabd78cc4b054135f
Change-Id: I1a7a8dd7870a71df9f0e7c30392cd9a23b5cab3e
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3126820
Commit-Queue: Brice Dobry <brice.dobry@futurewei.com>
Reviewed-by: Brice Dobry <brice.dobry@futurewei.com>
Cr-Commit-Position: refs/heads/main@{#76630}
---
 src/baseline/riscv64/baseline-assembler-riscv64-inl.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/baseline/riscv64/baseline-assembler-riscv64-inl.h b/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
index 1bb78be1ebe..663462fdb56 100644
--- a/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
+++ b/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
@@ -511,6 +511,11 @@ void BaselineAssembler::EmitReturn(MacroAssembler* masm) {
 
 #undef __
 
+inline void EnsureAccumulatorPreservedScope::AssertEqualToAccumulator(
+    Register reg) {
+  assembler_->masm()->Assert(eq, AbortReason::kUnexpectedValue, reg,
+                             Operand(kInterpreterAccumulatorRegister));
+}
 }  // namespace baseline
 }  // namespace internal
 }  // namespace v8
-- 
2.35.1

